src/frontend/src/optimizer/plan_node/logical_expand.rs

```diff
@@ -94,14 +94,30 @@ impl PlanTreeNodeUnary for LogicalExpand {
                     .collect_vec()
             })
             .collect_vec();
-        let (mut mapping, new_input_col_num) = input_col_change.into_parts();
-        mapping.extend({
-            mapping
-                .iter()
-                .map(|p| p.map(|i| i + new_input_col_num))
-                .collect_vec()
-        });
-        mapping.push(Some(2 * new_input_col_num));
+
+        let old_out_len = self.schema().len();
+        let old_in_len = self.input().schema().len();
+        let new_in_len = input.schema().len();
```

Comment:
Why are these different? This part seems confusing to me.

Reply:
What does "old" and "new" mean here?

Reply:
> This part seems confusing to me.

Indeed, really confusing. I couldn't find why it's fixed after I fixed it. Finally I realized the logic here is exactly the same as previous one (just clearer), the difference is the value of `new_in_len`.

Reply:
Still don't get it.
Why are these different?
```
self.input().schema().len();
input.schema().len()
```

Seems like something should be changed here. Maybe not in this PR, because it was the same logic as previously. But we should fix it.

Reply:
Oh, this method is to replace the `self.input` with a new `input`, so `input != self.input()`